Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php

```diff
@@ -1530,13 +1529,17 @@ public function isInRemovedNodes(NodeData $nodeData)
      */
     protected function createQueryBuilder(array $workspaces)
     {
+        $workspacesNames = array_map(static function (Workspace $workspace) {
+            return $workspace->getName();
+        }, $workspaces);
+
         /** @var QueryBuilder $queryBuilder */
         $queryBuilder = $this->entityManager->createQueryBuilder();
 
         $queryBuilder->select('n')
             ->from(NodeData::class, 'n')
             ->where('n.workspace IN (:workspaces)')
-            ->setParameter('workspaces', $workspaces, Connection::PARAM_STR_ARRAY);
+            ->setParameter('workspaces', $workspaceNames);
```

Comment:
So `n.workspace` is a string of the name? We effectively duplicate the fact that `name` == `@ORM\Id` here.
Shouldn't it also work if only the `::PARAM_STR_ARRAY` type hint is removed?

Reply:
Yes, it does. But since we know what we have at hand, this "optimizes" what Doctrine ORM needs to do through the checks that were "removed" to fix a performance regression. That's why I chose this route…

Reply:
We could leave the parameter type hint in, for clarity…

Reply:
Okay, sounds reasonable. Let's just not forget that this place needs to be in sync with the workspace `@Id` column.